src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -135,14 +135,18 @@ internal void ReleaseConnection()
                 {
                     if (!_isClosed && _tcpClient != null)
                     {
-                        //free cbt buffer
-                        if (_channelBindingToken != null)
+                        try
                         {
-                            _channelBindingToken.Close();
+                            // Gracefully close the transmission channel
```

Comment:
Do you want to set the Socket to non-blocking before sending the Quit command?

Reply:
You mean `_tcpClient.Client.Blocking = false;`?

Reply:
Yes, that avoids blocking in Dispose on Socket.Send when the TCP send buffer is full.